sdk-server-resteasy/src/main/java/com/manywho/sdk/services/ServiceApplication.java

```diff
@@ -27,7 +29,12 @@ public void initialize(String packageName, boolean isHttp) {
 
         Reflections reflections = injector.getInstance(Reflections.class);
 
-        objects.addAll(createInstances(reflections.getTypesAnnotatedWith(Path.class)));
+        // Filter the appropriate version specific controllers
+        Set<Class<?>> controllers = 
+            reflections.getTypesAnnotatedWith(Path.class).stream()
+            .filter(c -> c.getName().endsWith(this.isV2 ? "V1" : "V2") == false).collect(Collectors.toSet());
```

Comment:
That condition is formulated in a seriously confusing way ;)
this.isV2 ? "V1" : "V2" ? (as normally would be the other way around)
And then == false?
Any chance this can be less confusing?
Wouldn't:
`c.getName().endsWith(this.isV2 ? "V2" : "V1")`
give the same result?

Reply:
No, a V2 service gets any classes that **don't** end in V1. Non version-specific classes don't have any kind of 'V' suffix so we need to make sure it includes those too. I've updated the comment to make this clearer